weaviate/collection/classes.py

```diff
@@ -0,0 +1,519 @@
+            if cls_field == "name":
+                ret_dict["class"] = _capitalize_first_letter(val)
+            if isinstance(val, Enum):
+                ret_dict[cls_field] = str(val.value)
+            elif isinstance(val, (bool, float, str, int)):
+                ret_dict[cls_field] = str(val)
+            else:
+                assert isinstance(val, ConfigModel)
+                ret_dict[cls_field] = val.to_dict()
+
+        return ret_dict
+
+
+class PropertyConfig(ConfigModel):
+    indexFilterable: Optional[bool] = None
+    indexSearchable: Optional[bool] = None
+    tokenization: Optional[Tokenization] = None
+    description: Optional[str] = None
+    moduleConfig: Optional[ModuleConfig] = None
+
+
+class Property(PropertyConfig, ConfigModel):
+    name: str
+    dataType: DataType
+
+    def to_dict(self) -> Dict[str, Any]:
+        ret_dict = super().to_dict()
+        ret_dict["dataType"] = [ret_dict["dataType"]]
+        return ret_dict
+
+
+class ReferenceProperty(ConfigModel):
+    name: str
+    reference_class_name: str
+
+    def to_dict(self) -> Dict[str, Any]:
+        ret_dict = super().to_dict()
+        ref_collection_name = self.reference_class_name[0].upper()
+        if len(self.reference_class_name) > 1:
+            ref_collection_name += self.reference_class_name[1:]
+        ret_dict["dataType"] = [ref_collection_name]
+        return ret_dict
+
+
+class CollectionConfig(CollectionConfigBase):
+    properties: Optional[List[Union[Property, ReferenceProperty]]] = None
+
+    def model_post_init(self, __context: Any) -> None:
+        collection_name = self.name[0].upper()
+        if len(self.name) > 1:
+            collection_name += self.name[1:]
+        self.name = collection_name
+
+    def to_dict(self) -> Dict[str, Any]:
+        ret_dict = super().to_dict()
+
+        if self.properties is not None:
+            ret_dict["properties"] = [prop.to_dict() for prop in self.properties]
+
+        return ret_dict
+
+
+class MetadataGet(BaseModel):
+    vector: bool = False
+    distance: bool = False
+    certainty: bool = False
+    score: bool = False
+    explain_score: bool = Field(alias="explainScore", default=False)
+    is_consistent: bool = Field(alias="isConsistent", default=False)
+
+    def _get_fields(self) -> Set[str]:
+        additional_props: Set[str] = set()
+        for field, value in self.model_fields.items():
+            enabled: bool = getattr(self, field)
+            if enabled:
+                name = value.alias if value.alias is not None else field
+                additional_props.add(name)
+        return additional_props
+
+    def to_graphql(self) -> str:
+        additional_props = self._get_fields()
+        if len(additional_props) > 0:
+            return "_additional{" + " ".join(additional_props) + "}"
+        else:
+            return ""
+
+    def to_rest(self) -> str:
+        return ",".join(self._get_fields())
+
+
+@dataclass
+class MetadataReturn:
```

Comment:
Do we need to export this? Users only recieve it, but they should never create one?